docs/tutorials-samples/targeting-core-for-new-libraries.md

```diff
@@ -0,0 +1,298 @@
+
+For example, in .NET Core you can use `HttpClient` in the `System.Net.Http` API to perform network operations over Http.  However, the .NET 4.0 Framework does not include `HttpClient`!  Instead, you may need to use `WebClient` from the `System.Net` assembly.
+
+First, the `project.json` file should look something like this:
+
+```
+{
+    "frameworks":{
+        "net40":{
+            "frameworkAssemblies": {
+                "System.Net":"",
+                "System.Text.RegularExpressions":""
+            }
+            
+        },
+        "dotnet55":{
+            "dependencies": {
+                "System.Runtime":"4.0.0-rc1-*",
+                "System.Net.Http": "4.0.1-beta-23409",
+                "System.Text.RegularExpressions": "4.0.11-beta-23409",
+                "System.Threading.Tasks": "4.0.11-beta-23409"
+            }
+        }
+    }
+}
+
+```
+
+Note that framework assemblies being used are explicitly referenced in the `net40` target, and NuGet references are also explictly listed in the `dotnet55` target.
+
+Next, your `using`s in your source file can be adjusted like this:
+
+```csharp
+#if NET40
+// This only compiles for non-.NET 4.0 targets
+using System.Net;
+#else
+// This compiles for all other targets
+using System.Net.Http;
+using System.Threading.Tasks;
+#endif
+```
+
+And further down in the source, you can use guards to use those libraries conditionally:
+
+```csharp
+    public class Library
+    {
+#if NET40
+        private readonly WebClient _client = new WebClient();
+        private readonly object _locker = new object();
+#else
+        private readonly HttpClient _client = new HttpClient();
+#endif
+
+#if NET40
+        // .NET 4.0 does not have async/await
+        public string GetDotNetCount()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+          
+            var uri = new Uri(url);
+            
+            string result = "";
+            
+            // Lock here to provide thread-safety.
+            lock(_locker)
+            {
+                result = _client.DownloadString(uri);
+            }
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"Dotnet Foundation mentions .NET {dotNetCount} times!";
+        }
+#else
+        // .NET 4.5+ can use async/await!
+        public async Task<string> GetDotNetCountAsync()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+            
+            // HttpClient is thread-safe, so no need to explicitly lock here
+            var result = await _client.GetStringAsync(url);
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"dotnetfoundation.orgmentions .NET {dotNetCount} times in its HTML!";
+        }
+#endif
+    }
+```
+
+And that's it!
```

Comment:
Good point.  Currently I've only been able to test a project compatible with DNX (e.g dnxcore, dotnetXX) because the testrunner for Xunit is written with a depdency on DNX.  I looked into a .NET 4.0-compatible testrunner for Xunit but haven't been able to get it to run yet.